fabric-network/test/impl/query/queryagent.js

```diff
@@ -0,0 +1,163 @@
+/**
+ * Copyright 2020 IBM All Rights Reserved.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+'use strict';
+
+const sinon = require('sinon');
+const expect = require('chai').expect;
+
+const Query = require('fabric-common/lib/Query');
+const QueryAgent = require('fabric-network/lib/impl/query/queryagent');
+
+describe('QueryAgent', () => {
+	let query;
+	const peer1 = {name: 'peer1'};
+	const peer2 = {name: 'peer2'};
+
+	const error1 = new Error('peer1-error');
+	const error2 = new Error('peer2-error');
+	error1.connection = {name: 'peer1'};
+	error2.connection = {name: 'peer2'};
+	const noresulterror = new Error('No responses returned for query');
+
+	const validResonses = {
+		responses: [
+			{
+				response: {status: 200, payload: Buffer.from('peer1')},
+				connection: {name: 'peer1'}
+			},
+			{
+				response: {status: 200, payload: Buffer.from('peer2')},
+				connection: {name: 'peer2'}
+			}
+		]
+	};
+	const validResonse = {
+		responses: [
+			{
+				response: {status: 200, payload: Buffer.from('peer1')},
+				connection: {name: 'peer1'}
+			}
+		]
+	};
+	const errorResonses = {
+		errors: [
+			error1,
+			error2
+		]
+	};
+	const errorResonse = {
+		errors: [
+			error1
+		]
+	};
+
+
+	beforeEach(() => {
+		query = sinon.createStubInstance(Query);
+	});
+
+	afterEach(() => {
+		sinon.restore();
+	});
+
+	describe('#constructor', () => {
+
+		it('no options', async () => {
+			const queryAgent = new QueryAgent({});
+			expect(queryAgent).to.exist;
+		});
+		it('with options', async () => {
+			const queryAgent = new QueryAgent({}, {something: 33});
+			expect(queryAgent).to.exist;
+		});
+		it('with timeout options', async () => {
+			const queryAgent = new QueryAgent({}, {timeout: 3});
+			expect(queryAgent.requestTimeout).to.equal(3000);
+		});
+	});
+
+	describe('#send', () => {
+		let queryAgent;
+
+		beforeEach(() => {
+			queryAgent = new QueryAgent(query);
+		});
+
+		it('queries a single peer', async () => {
+			await queryAgent.send([peer1]);
+
+			const expected = {
+				targets: [peer1],
+				requestTimeout: 3000
```

Comment:
A good specific unit test might be that the request timeout passed in the constructor is passed to the internal call to `query.send`. Other tests don't need to include the request timeout since that is not what they are testing.

Reply:
added to constructor test